fastcrypto-zkp/src/bn254/poseidon/mod.rs

```diff
@@ -229,11 +236,15 @@ mod test {
             "4123755143677678663754455867798672266093104048057302051129414708339780424023"
         );
 
-        assert!(to_poseidon_hash(to_bigint_arr(vec![
-            0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23,
-            24, 25, 26, 27, 28, 29, 30, 31, 32
-        ]))
-        .is_err());
+        assert_eq!(
+            to_poseidon_hash(to_bigint_arr(vec![
+                0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22,
+                23, 24, 25, 26, 27, 28, 29, 30, 31, 32
+            ]))
+            .unwrap()
+            .to_string(),
+            "15368023340287843142129781602124963668572853984788169144128906033251913623349"
```

Comment:
these test cases were generated consistent with snarkjs lib. should we use js to make sure this is also correct? 

also - can we make sure the zklogin is using the to_poseidon_hash where this throws error (as it never should happen?)

Reply:
Sure. I could add a `zklogin_poseidon` function which only allows up to 32 inputs? Then it's also made explicit that this function should always be compatible with the one found in the circuit?